osu.Game/Scoring/ScoreInfo.cs

```diff
@@ -242,6 +242,15 @@ public IEnumerable<HitResultDisplayStatistic> GetStatisticsForDisplay()
             }
         }
 
+        public ScoreInfo DeepClone()
+        {
+            var clone = (ScoreInfo)MemberwiseClone();
+
+            clone.Statistics = new Dictionary<HitResult, int>(clone.Statistics);
```

Comment:
It's like... Half-deep. Arrays such as mods are still changeable, I believe.

This is one of the reasons that traditionally `Clone()` (via `ICloneable`) does not define whether it's a shallow or a deep clone.

I don't know if I should block on this or not so I'll just roll with it.

Reply:
yeah, maybe we should go with something like `MidDepthClone` as the name *shrug*. i think it's fine until an issue arises. else we're just wasting effort.